lib/ecto/query/planner.ex

```diff
@@ -1737,7 +1737,31 @@ defmodule Ecto.Query.Planner do
         error! query, expr, "field `#{field}` in `#{kind}` is a virtual field in schema #{inspect schema}"
 
       true ->
-        error! query, expr, "field `#{field}` in `#{kind}` does not exist in schema #{inspect schema}"
+        suggestion_str =
+          case closest_suggestion(field, schema) do
+            nil ->
+              ""
+            suggestion ->
+             ", did you mean `#{suggestion}`?"
+          end
+
+        error! query, expr, "field `#{field}` in `#{kind}` does not exist in schema #{inspect schema}" <> suggestion_str
+    end
+  end
+
+  defp closest_suggestion(input, schema) do
+    input_string = Atom.to_string(input)
+
+    schema.__schema__(:fields)
+    |> Enum.map(fn field ->
+      {field, String.jaro_distance(input_string, Atom.to_string(field))}
+    end)
+    |> Enum.max(fn {_f1, s1}, {_f2, s2} -> s1 >= s2 end)
+    |> case do
+      {field, score} when score >= 0.8 ->
+        field
+      _ ->
+        nil
```

Comment:
What we do in Elixir is to show multiple suggestions (up to 5) above a threshold:

```elixir
      for({key, val} <- candidates,
          dist = String.jaro_distance(base, Atom.to_string(key)),
          dist >= @function_threshold,
          do: {dist, key, val})
      |> Enum.sort(&(elem(&1, 0) >= elem(&2, 0)))
      |> Enum.take(@max_suggestions)
      |> Enum.sort(&(elem(&1, 1) <= elem(&2, 1)))
```

Should we do the same here? Your call. :)

Reply:
This sound better! I'll try :)